adoc/admin-security-certificates.adoc

```diff
@@ -678,72 +678,183 @@ The kubelet client certificate are signed by kube-controller-manager and the kub
-
-kubectl annotate --overwrite deployment/oidc-gangway -n kube-system secret.reloader.stakater.com/reload=oidc-gangway-cert
-
-kubectl annotate --overwrite deployment/metrics-server -n kube-system secret.reloader.stakater.com/reload=metrics-server-cert
+helm install \
+    --name <RELEASE_NAME> \
+    --namespace cert-manager \
+     suse/reloader
+----
++
+Or if you have selected the Helm 3 alternative also see <<helm-tiller-install>>:
++
+[source,bash]
+----
+helm install <RELEASE_NAME> \
+    --namespace cert-manager \
+    --create-namespace \
+    suse/reloader
 ----
 
-. Upload the Kubernetes CA cert/key pair to Secret in the `kube-system` namespace:
+. Install cert-manager via helm chart:
 +
 [source,bash]
 ----
-kubectl create secret tls kubernetes-ca --cert=pki/ca.crt --key=pki/ca.key -n kube-system
+helm install \
+    --name <RELEASE_NAME> \
+    --namespace cert-manager \
+    --set global.leaderElection.namespace=cert-manager \
+    --set installCRDs=true \
+    suse/cert-manager
 ----
 +
+Or if you have selected the Helm 3 alternative also see <<helm-tiller-install>>:
++
+[source,bash]
+----
+helm install <RELEASE_NAME> \
+    --namespace cert-manager \
+    --create-namespace \
+    --set global.leaderElection.namespace=cert-manager \
+    --set installCRDs=true \
+    suse/cert-manager
+----
+
+* Cert-Manager CA Issuer Resource
++
+The cert-manager CA issuer is a {kube} resource that represents a certificate authority (CA), which can generate signed certificates by honoring certificate signing requests (CSR). Each cert-manager certificate resource requires one referenced issuer in the ready state to be able to honor CSR requests.
++
 [NOTE]
 ====
-If you want to use a custom trusted CA certificate/key to sign the certificate, upload to the secret resource.
+An `Issuer` is a namespaced resource, and it can not issue certificates to the certificate resources in other namespaces.
+
+If you want to create a single Issuer that can be consumed in multiple namespaces, you should consider creating a `ClusterIssuer` resource. This is almost identical to the Issuer resource, however, it is cluster-wide so it can be used to issue certificates in all namespaces.
+====
+
+* Cert-Manager Certificate Resource
++
+The cert-manager has a custom resource, Certificate, which can be used to define a requested x509 certificate which will be renewed and kept up to date by an Issuer or ClusterIssuer resource.
+
+==== Client Certificate Rotation
+
+[WARNING]
+====
+If you are running a cluster using cilium version before 1.6, the cilium data is stored in the ETCD cluster, not the custom resources (CR). {skuba} generates a client certificate to read/write the cilium date to the ETCD cluster and the client certificate will expire after 1 year. Please follow the below steps to use cert-manager to automatically renew the cilium client certificate.
+====
 
+. Check the {productname} cilium version before 1.6
```

Comment:
Do you mean before cilium addon upgrade, people need to check if their cilium is before 1.6, then set up the cert rotation beforehand?

Do we need to consider some statements for after cilium addon upgrade (1.5 -> 1.6) but cilium not working because the migrated cilium-secret does have expired certs? So that means not just check if cilium before 1.6, also need to check if the version is under 1.6 but the cluster not working?

Reply:
Before cilium < 1.6, cilium stores data into the ETCD cluster, cilium >= 1.6, cilium stores data into custom definition (CR).
So the admin _only_ needs to rotate the cilium client certificate if cilium < 1.6.

Do I need to add other sentences to make it more clear?

Reply:
It was clarified now! Thanks :+1: